File: JavaScript/프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    var answer = 0;
    var cnt = citations.length;
    citations.sort((a, b) => b - a);

    for (let i=0; i <= citations[0]; i++) {
        if (i <= cnt) answer = i;
        let temp = 0;
        if (citations.includes(i)) {
            citations.forEach((c) => c == i ? temp++ : temp);
            cnt -= temp;
        }
    }
    return answer;
}

File: Java/프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int temp;
        int answer = 0;
        int cnt = citations.length;
        
        List<Integer> c = new ArrayList<>();
        for (int i=0; i<citations.length; i++) {
            c.add(citations[i]);
        }        
        c.sort((a, b) -> b - a);

        for (int i=0; i <= c.get(0); i++) {
            temp = 0;
            if (i <= cnt) answer = i;
            if (c.contains(i)) {
                final int target = i;
                long count = c.stream().filter(e -> e == target).count();
                cnt -= count;
            }
        }
        return answer;
    }
}